server-spi-private/src/main/java/org/keycloak/broker/provider/IdentityProviderMapper.java

```diff
@@ -17,25 +17,31 @@
 
 package org.keycloak.broker.provider;
 
-import org.keycloak.models.IdentityProviderMapperModel;
-import org.keycloak.models.KeycloakSession;
-import org.keycloak.models.RealmModel;
-import org.keycloak.models.UserModel;
+import org.keycloak.models.*;
 import org.keycloak.provider.ConfiguredProvider;
 import org.keycloak.provider.Provider;
 import org.keycloak.provider.ProviderFactory;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 /**
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
 public interface IdentityProviderMapper extends Provider, ProviderFactory<IdentityProviderMapper>,ConfiguredProvider {
     String ANY_PROVIDER = "*";
+    Set<IdentityProviderSyncMode> DEFAULT_IDENTITY_PROVIDER_MAPPER_SYNC_MODES = new HashSet<>(Arrays.asList(IdentityProviderSyncMode.LEGACY, IdentityProviderSyncMode.IMPORT));
 
     String[] getCompatibleProviders();
     String getDisplayCategory();
     String getDisplayType();
 
+    default Set<IdentityProviderSyncMode> supportedSyncModes() {
```

Comment:
Perhaps it would be cleaner to do:
`boolean supportsSyncMode(IdentityProviderSyncMode syncMode)`